tests/test_commit_template.rs

```diff
@@ -180,12 +180,32 @@ fn test_log_customize_short_id() {
     let decl = "template-aliases.'format_short_id(id)'";
     let stdout = test_env.jj_cmd_success(
         &repo_path,
-        &["log", "--config-toml", &format!("{decl}='id.shortest()'")],
+        &[
+            "log",
+            "--config-toml",
+            &format!("{decl}='id.shortest().upper()'"),
+        ],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    @  Q test.user@example.com 2001-02-03 04:05:08.000 +07:00 6
+    │  (empty) first
+    o  Z 1970-01-01 00:00:00.000 +00:00 0
+       (empty) (no description set)
+    "###);
+
+    // Strings like those returned by `short()` can be uppercased
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &[
+            "log",
+            "--config-toml",
+            &format!("{decl}='id.short().upper()'"),
```

Comment:
Shouldn't `.lower()` be tested as well (for example on a literal)? Also, isn't that lacking a test for `.shortest().upper()` (and `.upper().lower()`)?

Reply:
Perhaps, using `-T template` is easier to test various combinations.

For string methods, we can also add a test in template_parser.rs, which should be cheaper than spawning jj process.

Reply:
I agree, at least on of the tests I added is extraneous. I'll try to move it somewhere, maybe in a day or two.

Reply:
I think I'll just add a test to `test_templater.rs`. It does spawn a lot of `jj` processes, but the tests fit in and if we want to call `jj` less we'd need to rewrite those tests anyway.